### DISCUSSION TITLE
Is `group:nodeJs` behavior intended? (package.json and docker related files are updated separately)

### DISCUSSION THREAD

**Discussion by konomae** [renovatebot/renovate#23441] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Wanted end result.

Single PR will be created for package.json and docker related files.

I would like to update CI environment at the same time, since I'm using the engines constraints in package.json.

### What you tried so far.

I'm using [`config:recommended`](https://docs.renovatebot.com/presets-config/#configrecommended).
It contains `group:recommended` which contains `group:nodeJs`.

https://docs.renovatebot.com/presets-config/#configrecommended
https://docs.renovatebot.com/presets-group/#grouprecommended
https://docs.renovatebot.com/presets-group/#groupnodejs

I'm using ~~`cimg:node`~~ `cimg/node` as a docker base image.
I think it matches package pattern `/node$` in `group:nodeJs`.

According to the description:

> https://docs.renovatebot.com/presets-group/#groupnodejs
> Group anything that looks like Node.js together so that it's updated together.

So I think it will group dependencies about node in package.json and docker related files.
But it won't.
(Since matchDatasources is specified, is it only valid for docker?)

I tried this with sample repository. Two separate PRs were created.
https://github.com/konomae/renovate-repro-202307/pull/1
https://github.com/konomae/renovate-repro-202307/pull/2

Is this behavior intended?

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
Yours is not a standard node image and does not match `/node$` so you will need to add it to the group yourself

**Reply by konomae** [reply]
Thank you for your reply!

I made a typo. 🙏 (`cimg:node` → `cimg/node`)
But it still doesn't match `/node$`?

**Reply by rarkins** [reply]
In that case I'm not sure why that is and we should debug it

**Reply by konomae** [reply]
I made another repository and copied `group:nodeJs` configuration.
https://github.com/konomae/renovate-repro-202307-2/

Then, I tried some configuration changes.

I don't know if this helps, but I'm sharing what I noticed.

- Remove `matchDatasources`
  - Not affected
- Add `groupName`
  - Not affected
- Remove `matchDatasources` and add `groupName`
  - Affected
  - And the result: https://github.com/konomae/renovate-repro-202307-2/pull/7
    - That's what I expected.

**Comment by a88zach** [comment]
@konomae your solution is the only solution that worked for me
```
 {
      "groupName": "Node.js",
      "commitMessageTopic": "Node.js",
      "matchDepNames": ["node"],
      "matchPackagePatterns": ["/node$"]
    }
```

results in:

| Package | Type | Update | Change |
|---|---|---|---|
| docker.io/node | final | minor | `20.11.1-alpine` -> `20.12.0-alpine` |
| [node](https://nodejs.org) ([source](https://togithub.com/nodejs/node)) |  | minor | `20.11.1` -> `20.12.0` |
| [node](https://nodejs.org) ([source](https://togithub.com/nodejs/node)) | engines | minor | [`20.11.1` -> `20.12.0`](https://renovatebot.com/diffs/npm/node/v20.11.1/v20.12.0) |

**Reply by a88zach** [reply]
The above configuration stopped working.  Here is an updated configuration that does seem to work now
```
{
      "groupName": "Node.js",
      "commitMessageTopic": "Node.js",
      "matchDepNames": ["node", "docker.io/node"],
      "matchPackageNames": ["/(?:^|/)node$/", "!@nx/node"]
}
```
